### PR TITLE
refactor: add unlock checks for notification related controllers

### LINKED ISSUE #26064: [Bug]: Confusing popup / badge behavior (develop and latest 12.0.0 build)
### Describe the bug

1. There is one `Pending` transaction in the Activity list. 
2. After reloading / turning the extension OFF->ON popup appears automatically
3. Blue badge appears (I'm not sure if it should be shown because it is not an `Unapproved` transaction, it was confirmed by a user) 
4. Counter shows 3
5. After unlocking the wallet the blue badge disappears 

### Expected behavior

-

### Screenshots/Recordings

https://github.com/user-attachments/assets/beccadf3-3e54-4b02-9191-9580c7b75d9d

### Steps to reproduce

1. Have a transaction in `Pending` state
2. Reload the extension
3. Notice that popup opens automatically and blue badge is shown on MetaMask icon

### Error messages or log output

_No response_

### Detection stage

In production (default)

### Version

v12.0.0 beta, develop

### Build type

None

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### Severity

_No response_

### PR DESCRIPTION
**Pull request by Prithpal-Sooriya**
## **Description**

Ensures that we correctly call the preinstalled snap when the wallet is unlocked. Also prevents unlock confirmations being created due to snap calls when wallet is locked

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/26189?quickstart=1)

## **Related issues**

[Ticket](https://consensyssoftware.atlassian.net/browse/NOTIFY-865)

Fixes: https://github.com/MetaMask/metamask-extension/issues/26064

## **Manual testing steps**

1. Install the extension and enable notifications
2. Restart the browser, or reload the extension
3. Should see no auto opening of the wallet, and no unlock confirmations being made.

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/scripts/controllers/authentication/authentication-controller.ts [hunk 1]
```diff
@@ -105,6 +116,25 @@ export default class AuthenticationController extends BaseController<
 > {
   #metametrics: MetaMetricsAuth;
 
+  #isUnlocked = false;
+
+  #keyringController = {
+    setupLockedState: () => {
```
Comment (mathieuartu) [inline app/scripts/controllers/authentication/authentication-controller.ts:122]: Same comment as in core, maybe for readability:

```suggestion
    setupLockedStateSubscriptions: () => {
```

**Review by mathieuartu — DISMISSED** [review]

**Review by danjm — DISMISSED** [review]

File: app/scripts/controllers/authentication/authentication-controller.ts (hunk 1 above)
Comment (Prithpal-Sooriya) [reply app/scripts/controllers/authentication/authentication-controller.ts:122]: As we need to add tests for test coverage, I'll update to this.

**Comment by Prithpal-Sooriya** [comment]
Unfortunately there are some test coverage that needs to be added, I'll re-ping you again once test coverage has been met

**Review by mathieuartu — APPROVED** [review]

**Review by matteoscurati — APPROVED** [review]

**Comment by Gudahtt** [comment]
Please use the title prefix `fix:` for bug fixes in the future. This PR fixes a bug and changes how the application behaves, so it is not a refactor.

File: app/scripts/controllers/metamask-notifications/metamask-notifications.ts [hunk 2]
```diff
@@ -210,25 +213,25 @@ export type AllowedActions =
...
 
 // Events
-export type MetamaskNotificationsControllerMessengerEvents =
-  ControllerStateChangeEvent<
-    typeof controllerName,
-    MetamaskNotificationsControllerState
-  >;
+export type Events =
+  | MetamaskNotificationsControllerNotificationsListUpdatedEvent
+  | MetamaskNotificationsControllerMarkNotificationsAsRead;
```
Comment (Prithpal-Sooriya) [inline app/scripts/controllers/metamask-notifications/metamask-notifications.ts:218]: These events are used for a  feature only available in v12.0.1+, not in v12.0.0.

This is just some cleanup to separate out the controller events vs external events.

**Pull request merged**